pkg/apis/ceph.rook.io/v1/object.go

```diff
@@ -36,6 +36,15 @@ func (s *ObjectStoreSpec) IsTLSEnabled() bool {
 	return s.Gateway.SecurePort != 0 && (s.Gateway.SSLCertificateRef != "" || s.GetServiceServingCert() != "")
 }
 
+func (s *ObjectStoreSpec) GetPort() (int32, error) {
```

Comment:
@BlaineEXE : One more request which I want to make(out of this PR scope).  Please replace [`setObjectStorePort()`](https://github.com/rook/rook/blob/c961d9f7acebf4369dfe233b453c248fff0c52f7/pkg/operator/ceph/object/bucket/provisioner.go#L543) with this function, currently that function chose port from `rgw svc` and picks the first port, Other part of the code prefers for TLS, if it is enabled so for OBC's we end up having `https` endpoint with non secure port.

Reply:
I don't fully understand the concern. It sounds like the funcionality for `setObjectStorePort()` was implemented incorrectly. Perhaps a bug should be filed and that issue fixed separately.

Reply:
Created #8141 for this.

Reply:
Okay. I will wait this PR to merge, then fix that issue. Thanks for opening up the issue